core/src/main/java/org/apache/iceberg/GenericManifestFile.java

```diff
@@ -145,7 +145,12 @@ private GenericManifestFile(GenericManifestFile toCopy) {
     this.existingRowsCount = toCopy.existingRowsCount;
     this.deletedFilesCount = toCopy.deletedFilesCount;
     this.deletedRowsCount = toCopy.deletedRowsCount;
-    this.partitions = copyList(toCopy.partitions, PartitionFieldSummary::copy);
+    if (toCopy.partitions != null) {
+      this.partitions = copyList(toCopy.partitions(), PartitionFieldSummary::copy)
+          .toArray(new PartitionFieldSummary[0]);
```

Comment:
Do we need to use `copyList`? What about using a Stream instead?

```java
Streams.of(toCopy.partitions).map(PartitionFieldSummary::copy).toArray(PartitionFieldSummary[]::new);
```

Reply:
I think you are saying that we don't have to convert the `partitions` array to a list then copy it again , and convert it back to `partitions` array in the end ( if use the `copyList` approach`). 

Yes, I agree.